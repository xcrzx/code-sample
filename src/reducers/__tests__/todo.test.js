import {
  toggleTodo,
  fetchTodosRequest,
  fetchTodosResponse,
} from 'actions/todo'
import reducer, { defaultState } from 'reducers/todo'

const sampleResponse = {
  entities: {
    0: {
      completed: true,
    },
  },
}

describe('Todo reducer', () => {
  it('Handles all actions', () => {
    expect(reducer(defaultState, fetchTodosRequest())).toMatchSnapshot()
    expect(reducer(defaultState, fetchTodosResponse(sampleResponse))).toMatchSnapshot()
    expect(reducer(defaultState, fetchTodosResponse(Error('error')))).toMatchSnapshot()

    expect(reducer(sampleResponse, toggleTodo(0))).toMatchSnapshot()
  })
})
