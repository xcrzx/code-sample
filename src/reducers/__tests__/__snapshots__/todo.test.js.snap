// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo reducer Handles all actions 1`] = `
Object {
  "entities": Object {},
  "isFetching": true,
}
`;

exports[`Todo reducer Handles all actions 2`] = `
Object {
  "entities": Object {
    "0": Object {
      "completed": true,
    },
  },
  "isFetching": false,
}
`;

exports[`Todo reducer Handles all actions 3`] = `
Object {
  "entities": Object {},
  "isFetching": false,
}
`;

exports[`Todo reducer Handles all actions 4`] = `
Object {
  "entities": Object {
    "0": Object {
      "completed": false,
    },
  },
}
`;
